### DISCUSSION TITLE
Parameterized yaml files

### DISCUSSION THREAD

**Discussion by chefcook** [datreeio/datree#735] [Q&A] [answered]
Hello together, 

I would like to know if there is a possibility to test parametrized YAMLs.

I cannot get my YAML tested because I constantly run into a YAML validation error.

`yaml validation error: yaml: invalid map key: map[string]interface {}{"NAMESPACE":interface {}(nil)}`

My YAMLs are parametrized with handlebars. 

Here is an example.

![image](https://user-images.githubusercontent.com/12730470/180182343-1b849f0f-17f6-4356-ba68-179831ecbd7e.png)

Am I missing something in the rules? I have not found a rule where I could tweak something like this.

**Comment by adifayer** [comment, marked as answer]
@chefcook What templating tool solution do you use, helm?

**Reply by chefcook** [reply]
Yes I use Helm for templating generally speaking. But I still don't understand the connection with my described error.

**Reply by adifayer** [reply]
I think the solution for this one is to first create the manifest (inject the variable values), and then pipe the output to Datree test command, similar to:
`[created_manifest] | datree test -`

**Reply by chefcook** [reply]
Thanks for the feedback. I have it basically solved about as you described.

"fake" deployment into a local directory. So all variables are filled and then run datree over it.

**Reply by eyarz** [reply]
@chefcook you can also use the helm plugin to do it (generating the manifests and running Datree) for you:
https://github.com/datreeio/helm-datree
